src/ExtInterface/ReefMod/Domain.jl

```diff
@@ -309,12 +309,27 @@ function load_initial_cover(::Type{ReefModDomain}, data_path::String, loc_ids::V
         icc_data[:, :, i] = Matrix(CSV.read(fn, DataFrame; drop=[1], header=false, comment="#"))
     end
 
+    # use ReefMod distribution for coral size class population (shape parameters have units log(cm^2))
+    # as suggested by YM (pers comm. 2023-08-08 12:55pm AEDT)
+    reef_mod_area_dist = LogNormal(log(700), log(4))
+    # get bin edges for coral diameters and transform to coral areas in cm^2
+    bin_edges_area = colony_mean_area(Float64[0, 2, 5, 10, 20, 40, 80])
+
+    # find integral density between bounds of each size class areas
+    diff_cdf_integral = cdf.(reef_mod_area_dist, bin_edges_area[2:end]) .- cdf.(reef_mod_area_dist, bin_edges_area[1:end-1])
+    # normalise densities for each size class to create weightings
+    size_class_weights = diff_cdf_integral ./ sum(diff_cdf_integral)
+
     # Take the mean over repeats, as suggested by YM (pers comm. 2023-02-27 12:40pm AEDT)
     # Convert from percent to relative values
-    icc_data = dropdims(mean(icc_data, dims=2), dims=2) ./ 100.0
+    icc_data = ((dropdims(mean(icc_data, dims=2), dims=2)) ./ 100.0)
+
+    # Repeat species over each size class and reshape to give ADRIA compatible size (36 * nsites).
+    # Multiply by size class weights to give initial cover distribution over size class.
+    icc_data = reshape(repeat(icc_data, 6, 1), length(loc_ids), length(icc_files) * 6)' .* repeat(size_class_weights, 6)
```

Comment:
Agree this isn't the best approach, but I'll leave it to you to tidy up.

Is it not possible to apply the weights with L 325?

Reply:
I tried doing this previously but you need the repeats of the coral species so that you get each of the size classes for each coral species, otherwise you just get one size class for each species. I could move the repeat up and then the final line is just reshaping and multiplying by weights...

Reply:
So `icc_data` and `size_class_weights` are not of compatible shapes?

Reply:
icc data is n_locsX6 and size_class_weights is 6X1, but if you just multiply them together you get (species x* size class x) but not (species x * size class y). You need all permutations of  (species x * size class y), for each location, to get the final 36 species/size class dimension.

Reply:
How is `size_class_weights` of shape 6x1? Is it not a vector?

In any case:

```julia
n_locs = 50
n_species = 6
n_sc = 6
icc_data = rand(n_locs , n_species)  # 50 example locations
size_class_weights = rand(n_sc)

# Multiply each row of icc_data with weights with each element in size_class_weights
# resulting in a matrix (shape: 50 x 6) where each entry is a vector
# You could replace icc_data instead of assigning to x
x = icc_data .* [size_class_weights]

# You can verify correctness with:
all(icc_data[1, 1] .* size_class_weights .== x[1,1])

# Join vectors for each row and create matrix
Matrix(transpose(hcat(reduce.(vcat, eachrow(x))...)))

# ... or with pipes:
# hcat(reduce.(vcat, eachrow(x))...) |> transpose |> Matrix
```

Is that what you want?

Reply:
Thanks Takuya, that's perfect. I didn't realise elementwise multiplication with square brakets gives a deconstructed dot product. I'll implement this and then it's ready for review.

Reply:
Ready for review :)